src/Microsoft.ML.Data/DataView/EmptyDataView.cs

```diff
@@ -59,12 +59,7 @@ public Cursor(IChannelProvider provider, Schema schema, Func<int, bool> needCol)
 
             public override ValueGetter<RowId> GetIdGetter()
             {
-                return
-                    (ref RowId val) =>
-                    {
-                        Ch.Assert(!IsGood);
-                        throw Ch.Except("Cannot call ID getter in current state");
-                    };
+                return (ref RowId val) => throw Ch.Except("Cannot call ID getter in current state");
```

Comment:
>"Cannot call ID getter in current state" [](start = 58, length = 40)

In DataViewConstructionUtils you change it to RowCursorUtils.FetchValueStateError, but remain message same here.

Reply:
Yes, and I see the same on the getter below as well. A more thorough review of implementations of this specifically revealed a few more, thank you @Ivanidzo4ka.